File: Detecting Duplicate Parentheses in an Expression/Detecting_Duplicate_Parentheses_in_an_Expression.cpp
#include <iostream>
#include <stack>
using namespace std;

bool findDuplicateParentheses(string expression) {
    stack<char> st;
    for (char c : expression) {
        if (c == ')') {
            int insideCount = 0;
            while (!st.empty() && st.top() != '(') {
                st.pop();
                insideCount++;
            }
            if (insideCount < 1) {
                return true;
            }
            st.pop(); // Remove the matching '('
        } else {
            st.push(c);
        }
    }
    return st.empty();
}

int main() {
    string expression = "(((a+(b))+(c+d)))";
    if (findDuplicateParentheses(expression)) {
        cout << "Duplicate Found" << endl;
    } else {
        cout << "No Duplicates Found" << endl;
    }
    return 0;
}

File: Detecting Duplicate Parentheses in an Expression/Detecting_Duplicate_Parentheses_in_an_Expression.py
def find_duplicate_parentheses(expression):
    stack = []
    for char in expression:
        if char == ')':
            inside_count = 0
            while stack and stack[-1] != '(':
                stack.pop()
                inside_count += 1
            if inside_count < 1:
                return True
            stack.pop()  # Remove the matching '('
        else:
            stack.append(char)
    return not stack

# Example usage
expression = "(((a+(b))+(c+d)))"
if find_duplicate_parentheses(expression):
    print("Duplicate Found")
else:
    print("No Duplicates Found")
